### PR TITLE
Separate Site Verification and Search Console setup steps in the wizard components.

### PR DESCRIPTION
**Pull request by lucymtc**
## Summary
Moves Search Console setup steps from Site Verification component to the actual Search Console wizard step component.

Addresses issue https://github.com/google/site-kit-wp/issues/359

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/components/setup/search-console.js [hunk 1]
```diff
@@ -48,24 +48,29 @@ class SearchConsole extends Component {
...
 		const { isAuthenticated, shouldSetup } = this.props;
 
-		if ( isAuthenticated && shouldSetup ) {
-			this.requestSearchConsoleSiteList();
+		if ( ! isAuthenticated || ! shouldSetup ) {
+			return;
 		}
-	}
 
-	shouldComponentUpdate( nextProps ) {
```
Comment (lucymtc) [inline assets/js/components/setup/search-console.js]: Removed `shouldComponentUpdate` as runs `requestSearchConsoleSiteList()` and makes unnecessary requests to `matched-sites` and `insert` which have already been requested.

File: assets/js/components/setup/search-console.js [hunk 2]
```diff
@@ -134,6 +139,37 @@ class SearchConsole extends Component {
...
+	async saveProperty( siteURL ) {
+		try {
+			const responseData = await data.set( 'modules', 'search-console', 'save-property', { siteURL } );
+
+			return new Promise( ( resolve ) => {
+				resolve( responseData );
+			} );
+		} catch ( err ) {
+			throw err;
+		}
```
Comment (aaemnnosttv) [inline assets/js/components/setup/search-console.js]: This is an unnecessary try/catch block since it does nothing else other than re-throwing the error. If there's nothing else to do in this case, it should be removed.

File: assets/js/components/setup/search-console.js [hunk 3]
```diff
@@ -134,6 +139,37 @@ class SearchConsole extends Component {
...
+		try {
+			const responseData = await data.get( 'modules', 'search-console', 'is-site-exist' );
+
+			return new Promise( ( resolve ) => {
+				resolve( responseData );
+			} );
+		} catch ( err ) {
+			throw err;
+		}
+	}
```
Comment (aaemnnosttv) [inline assets/js/components/setup/search-console.js]: Same as above.

File: assets/js/components/setup/site-verification.js [hunk 4]
```diff
@@ -146,18 +101,6 @@ class SiteVerification extends Component {
...
-		try {
-			data.set( 'modules', 'search-console', 'insert', { siteURL } );
-		} catch ( err ) {
-			throw err;
-		}
-	}
-
 	async insertSiteVerification( siteURL ) {
 		try {
 			const { shouldSetup } = this.props;
```
Comment (felixarntz) [inline assets/js/components/setup/site-verification.js]: The check for `shouldSetup` can be removed from here. It is already checked early in the component's lifecycle.

The `try catch` block can also be removed, similar to the cases mentioned above by @aaemnnosttv.

File: assets/js/components/setup/search-console.js [hunk 5]
```diff
@@ -134,6 +139,37 @@ class SearchConsole extends Component {
...
 
+	/**
+	 * Save Search Console property.
+	 * @param { string } siteURL
+	 */
+	async saveProperty( siteURL ) {
+		try {
+			const responseData = await data.set( 'modules', 'search-console', 'save-property', { siteURL } );
+
+			return new Promise( ( resolve ) => {
```
Comment (felixarntz) [inline assets/js/components/setup/search-console.js]: We can probably remove this wrapping promise as well.

File: assets/js/components/setup/search-console.js [hunk 6]
```diff
@@ -134,6 +139,37 @@ class SearchConsole extends Component {
...
+	}
+
+	/**
+	 * Check whether a property for the site exists in Search Console.
+	 */
+	async isSiteExist() {
+		try {
+			const responseData = await data.get( 'modules', 'search-console', 'is-site-exist' );
+
+			return new Promise( ( resolve ) => {
```
Comment (felixarntz) [inline assets/js/components/setup/search-console.js]: Same as above.

File: assets/js/components/setup/search-console.js [hunk 7]
```diff
@@ -181,11 +146,6 @@ class SearchConsole extends Component {
 			try {
 				await this.insertPropertyToSearchConsole( siteURL );
 
-				this.setState( {
-					loading: false,
-					connected: true,
-				} );
```
Comment (felixarntz) [inline assets/js/components/setup/search-console.js]: Why did you remove this?

Comment (lucymtc) [reply assets/js/components/setup/search-console.js]: @felixarntz because state is already set in `insertPropertyToSearchConsole()` https://github.com/google/site-kit-wp/blob/010cfb9e1e01d3aab790e6dfb9656e05a26fefd2/assets/js/components/setup/search-console.js#L129-L136
